### ISSUE TITLE
Lint/ShadowingOuterLocalVariable: a false positive case with if/else statement and block variable

### ISSUE THREAD

**Issue by deepj** [rubocop/rubocop#10479]
## Expected behavior

No offense with `Lint/ShadowingOuterLocalVariable` cop

## Actual behavior

```ruby
W: Lint/ShadowingOuterLocalVariable: Shadowing outer local variable - email.
      list.each.map do |email|
```

`email` variable is not declared on any previous level.

## Steps to reproduce the problem

Code snippet

```ruby
  def emails_list(list, separator: ', ', short: false)
    return if list.blank?

    if short
      email = list.first
      out   = mail_to(h(email))
      out << ', &hellip;'.html_safe
    else
      list.each.map do |email|
        escaped = h(email)
        mail_to escaped, escaped, target: '_blank'
      end.uniq.join(separator).html_safe
    end
  end
```

```
rubocop --only Lint/ShadowingOuterLocalVariable
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.26.1 (using Parser 3.1.1.0, rubocop-ast 1.16.0, running on ruby 3.1.1 arm64-darwin21)
  - rubocop-performance 1.13.3
  - rubocop-rails 2.14.2
```

**Comment by stefanoc** [comment]
It looks like this bug is still triggered if you add an additional `elsif` clause to the example:

```
  def emails_list(list, separator: ', ', short: false)
    return if list.blank?

    if short
      email = list.first
      out   = mail_to(h(email))
      out << ', &hellip;'.html_safe
    elsif whatever
      # noop
    else
      list.each.map do |email|
        escaped = h(email)
        mail_to escaped, escaped, target: '_blank'
      end.uniq.join(separator).html_safe
    end
  end
```

```
 %% rubocop --only Lint/ShadowingOuterLocalVariable test.rb
Inspecting 1 file
W

Offenses:

test.rb:11:25: W: Lint/ShadowingOuterLocalVariable: Shadowing outer local variable - email.
      list.each.map do |email|
                        ^^^^^

1 file inspected, 1 offense detected

%% rubocop -V
1.35.1 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2 x86_64-darwin21)
```

**Comment by stefanoc** [comment]
@koic let me know if I should I open a new issue

**Comment by koic** [comment]
@stefanoc Yeah, can you open a new issue?
